src/constant.h

```diff
@@ -7,9 +7,12 @@
 // Global constants used by Moria
 // clang-format off
 
-#define MAX_UCHAR         255
-#define MAX_SHORT       32767   // maximum short/long signed ints
-#define MAX_LONG   0x7FFFFFFFL
+#include <limits>
+#include <cstdint>
+
+constexpr unsigned char MAX_UCHAR = std::numeric_limits<unsigned char>::max();
```

Comment:
Perhaps you could update the `RNG_M` in `rnd.cpp` to also use the `numeric_limits`?